dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/logger/LogcatLogHandler.kt

```diff
@@ -47,45 +42,38 @@ internal class LogcatLogHandler(
         }
     }
 
-    private fun resolveTag(): String {
-        return if (Datadog.isDebug) {
-            return resolveTagFromCallerClassName()
-        } else {
-            serviceName
-        }
-    }
-
-    private fun resolveTagFromCallerClassName(): String {
-        val className: String = tryAndSearchClassName() ?: return serviceName
-        var tag = stripAnonymousPart(className)
-        tag = tag.substring(tag.lastIndexOf('.') + 1)
+    // endregion
 
-        return sanitizeTag(tag)
-    }
+    // region Internal
 
-    private fun tryAndSearchClassName(): String? {
-        val stackTrace = Throwable().stackTrace
-        // it might happen that when called from Java code the stack index to be ++
-        // due to the Java - Kotlin bridge method so we need an extra check here.
-        for (i in callerNameStackIndex until stackTrace.size) {
-            val className = stackTrace[i].className
-            if (className != LOGGER_CLASS_NAME)
-                return className
+    private fun resolveTag(stackTraceElement: StackTraceElement?): String {
+        return if (stackTraceElement == null) {
+            serviceName
+        } else {
+            stackTraceElement.className
+                .replace(ANONYMOUS_CLASS, "")
+                .substringAfterLast('.')
```

Comment:
Hmmm I think you forgot to add the check for below Android.N to have a limited amount for the TAG length. I think this will crash otherwise in those versions.